### DISCUSSION TITLE
COG driver not found

### DISCUSSION THREAD

**Discussion by charalamm** [davidfrantz/force#272] [Q&A] [answered]
Hello, 

I am trying to use [this](https://hub.docker.com/layers/davidfrantz/force/latest/images/sha256-53aa8534dee95df71bef83baf21fb5c9b6dd695509bb74ac6ceab5ebdf5ad604?context=explore) force docker image to produce some COG files with `force-level2` but I am getting the following error in the logs.

Also, running the container as root and upgrading GDAL to version `GDAL 3.3.2, released 2021/09/01` (that is the futhest I got) produces a new error about a DEBUG environment variable and asks for recompiling force.

Does anybody have any ideas on how to proceed with this error?

Thank you!

The log file:
```
/data/sentinel2_level1/T32UPG/S2B_MSIL1C_20220903T101559_N0400_R065_T32UPG_20220903T122620.SAFE
: dc:  68.11%. wc:  21.61%. sc:   0.13%. cc:  66.58%. AOD: 0.1825. # of targets: 197/818. COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
COG driver not found
error in cubing Level 2 products
Tiling images failed! Error in geometric module.
```

**Comment by davidfrantz** [comment, marked as answer]
Hi @charalamm,

yes, COG is only available from GDAL >= 3.1. Our Docker container still runs on 3.0.4. 
Note that updating the GDAL version often breaks things. This is why I stick to versions that are rigorously tested.

However, the designated COG driver is only a convenient shortcut for producing COGs. They can also be generated using GeoTiff driver and some specific configuration options, please see here:
https://trac.osgeo.org/gdal/wiki/CloudOptimizedGeoTIFF

Cheers,
David

PS: if you really want to upgrade GDAL, it would be best to update the base image (https://github.com/davidfrantz/base_image), and then use this as basis for building a new FORCE. However, as said, accidents might happen as this GDAL version was not tested..

**Comment by charalamm** [comment]
Hello @davidfrantz! So, if I understand correctly there is no way to create COGs with FORCE without side effects since it is fully compatible only with GDAL 3.0.4. Even running FORCE without docker would require GDAL 3.0.4. Right?

**Reply by davidfrantz** [reply]
Not quite. It is guaranteed to be fully compatible with 3.0.4. It may, or may not work with higher versions.

And as written, COGs can also be generated with 3.0.4, but not by simply naming the COG driver, but by using GeoTiff driver with specific output options (a COG is simply a specially configured GeoTiff).
